### DISCUSSION TITLE
How to Change the Text Trait Events

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#5572] [General]
By Default the Text Trait Events runnning in onblur method i am trying to the the method on onChange so the value is updated in every keystroke

```
editor.TraitManager.addType('coupon-code-text', {
    extends: 'text',
    events: {
      input: 'input',
    },
    onValueChange: function () {
      const value = this.model.get('value');

      const component = this.target;

      if (component) {
        component.components(value);
      }
    },
  });
```

The Events property is not working as what i expected by default the text trait event it onBlur

**Comment by artf** [comment]
@RakulAgn can you please read the docs https://grapesjs.com/docs/modules/Traits.html#define-new-trait-type you're not using the right API
